whois-api/src/main/java/net/ripe/db/whois/api/rdap/RedactionObjectMapper.java

```diff
@@ -0,0 +1,150 @@
+package net.ripe.db.whois.api.rdap;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import net.ripe.db.whois.api.rdap.domain.Redaction;
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.apache.commons.compress.utils.Lists;
+
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static net.ripe.db.whois.api.rdap.RdapObjectMapper.CONTACT_ATTRIBUTE_TO_ROLE_NAME;
+import static net.ripe.db.whois.common.rpsl.AttributeType.E_MAIL;
+import static net.ripe.db.whois.common.rpsl.AttributeType.NOTIFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MBRS_BY_REF;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_DOMAINS;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_LOWER;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_REF;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_ROUTES;
+
+@Component
+public class RedactionObjectMapper {
+
+    private final RpslObjectDao rpslObjectDao;
+    private static final Map<AttributeType, Pair> UNSUPPORTED_ENTITIES = Maps.newHashMap();
+    static {
+        UNSUPPORTED_ENTITIES.put(MBRS_BY_REF, new Pair("Authenticate members by reference", "No registrant mntner"));
+        UNSUPPORTED_ENTITIES.put(MNT_DOMAINS, new Pair("Authenticate domain objects", "No registrant mntner"));
+        UNSUPPORTED_ENTITIES.put(MNT_LOWER, new Pair("Authenticate more specific resources", "No registrant mntner"));
+        UNSUPPORTED_ENTITIES.put(MNT_REF, new Pair("Authenticate incoming references", "No registrant mntner"));
+        UNSUPPORTED_ENTITIES.put(MNT_ROUTES, new Pair("Authenticate route objects", "No registrant mntner"));
+    }
+
+    private static final Map<AttributeType, Pair> UNSUPPORTED_VCARDS = Maps.newHashMap();
+    {
+        //UNSUPPORTED_VCARDS.put(E_MAIL, new Pair("e-mail contact information", "Personal data"));
+        UNSUPPORTED_VCARDS.put(NOTIFY, new Pair("Updates notification e-mail information", "Personal data"));
+    }
+
+    public static String UNSUPPORTED_ENTITIES_SYNTAX = "$.entities[?(@.handle=='%s')]";
+
+    public static String UNSUPPORTED_VCARD_SYNTAX = "$.entities[?(@.roles=='%s')].vcardArray[1][?(@[0]=='%s')]";
+
+    RedactionObjectMapper(final RpslObjectDao rpslObjectDao){
+        this.rpslObjectDao = rpslObjectDao;
```

Comment:
I think instead hardcoding prepath ourself we should use the library as you anyways adding the dependency.